interchange/DeviceResources.capnp

```diff
@@ -656,6 +657,46 @@ struct Device {
     lutElements @1 : List(LutElements);
   }
 
+  ######################################
+  # BEL chains definitions
+  ######################################
```

Comment:
Do you have any information about why this is needed? This all seems easily detectable from the routing?

Reply:
I’m not sure if I understand it correctly. The idea is to have a set of chain definitions like carries and any other chainable BELs in the schema to provide information to PnR tool which cells can be clusterized and placed as a cluster/chain.

AFAIK it is possible to detect chained cells by checking if e.g. cell has a driver which is another cell of the same type and iterate until we hit a constant driver. If this is what you meant then it is doable, but this approach would probably take more time to detect them rather than providing in schema which cell types are chainable, picking the right ones and clustering them together.